Fixes for duplicated othernodes, date/time logging for crashcloudwatch and buffer_size for init-quorum

ExcChua opened the pull request:
Added datetime to crashcloudwatch and exitcodes to quorum, so that crashquorum doesn't emit a cloudwatch when quorum is stopped by supervisorctl.

ExcChua committed:
Update crashcloudwatch.py

ExcChua committed:
Updated import statement and references to datetime

Updated import statement and references to datetime

ExcChua committed:
Updated init-quorum.sh with buffer_size

Updated init-quorum.sh with buffer_size=1024 for both quorum and constellation
Also updated quorum with exitcodes=0,1,2

ExcChua committed:
Prevented duplicated othernodes

Prevent duplicated othernodes in complete_constellation_config

ExcChua commented:
Prevent duplicated othernodes in init-bootnodes.sh
Updated init-quorum.sh with buffer_size
Updated crashcloudwatch.py to log with date/time.

Lsquared13 commented on packer/instance-scripts/crashcloudwatch.py:
```diff
@@ -102,7 +103,7 @@ def runforever(self, test=False):
                 childutils.listener.ok(self.stdout)
                 if test:
-                    self.stderr.write('non-exited event\n')
+                    self.stderr.write(datetime.now().strftime("%Y-%m-%d_%H:%M:%S") + ' non-exited event\n')
```
nit: Throw the date template string into a global variable at the top

Lsquared13 commented on packer/instance-scripts/init-bootnode.sh:
```diff
@@ -20,25 +20,32 @@ function complete_constellation_config {
-    do
-        local NUM_BOOTNODES=$(cat /opt/quorum/info/bootnode-counts/${region}.txt)
-        for index in $(seq 0 $(expr $NUM_BOOTNODES - 1))
+    DIR=$(dirname "${CONSTELLATION_CONFIG_PATH}")
```
Make these variables local please. If you do need access to it outside the function, create the variable in the main code path and pass it to the function as an argument.

ExcChua replied:
After making these changes, do I do a new PR, or what? No idea how to proceed.

Lsquared13 replied:
See main thread comment below.

Lsquared13 reviewed (changes requested):
Please confirm that you've run the private transaction test on a small single region network

Lsquared13 commented:
After addressing the changes, I think you just push to the source branch `EximChua:master` and it updates automatically (for that reason I recommend you don't make your PRs from master, just to make your life easier if you want to do more work on the repo).

If that doesn't work, ask some other folks who have actually done PRs regularly, but I think that's how it goes

ExcChua committed:
Made changes as suggested by Louis

Made YMDHMS a constant in crashcloudwatch and localized DIR and FLAGFILE in init-bootnode

ExcChua commented:
transaction test was ran on the network, and it worked.

ExcChua commented:
Merge completed.
